app/src/main/java/com/ubergeek42/WeechatAndroid/WeechatActivity.kt

```diff
@@ -199,7 +208,7 @@ class WeechatActivity : AppCompatActivity(), CutePageChangeListener,
 
         menuBackgroundDrawable = ContextCompat.getDrawable(this, R.drawable.bg_popup_menu)!!
 
-        if (P.isServiceAlive()) connect()
+        if (savedInstanceState == null && P.isServiceAlive()) connect()
```

Comment:
What's this about?

Reply:
Uhh not really sure, did that commit a while back

I'm assuming that `connect()` was called too often but this wasn't a problem since if we were connected it was no-op. Now that it checks for permissions and stuff it was easier to fix the too much part than reworking the logic otherwise?

Can't be sure tho